terraform/environments/core-shared-services/ad-fixngo.tf

```diff
@@ -0,0 +1,786 @@
+            ])
+          }
+        }
+        egress = {
+          all = {
+            port        = 0
+            protocol    = -1
+            cidr_blocks = ["0.0.0.0/0"]
+          }
+        }
+      }
+    }
+
+    tags = merge(local.tags, {
+      source-code            = "https://github.com/ministryofjustice/modernisation-platform"
+      infrastructure-support = "DSO:digital-studio-operations-team@digital.justice.gov.uk"
+    })
+  }
+}
+
+data "aws_ami" "ad_fixngo" {
+  for_each = toset(distinct([for key, value in local.ad_fixngo.aws_instances : value.ami_name]))
+
+  most_recent = true
+  owners      = ["self"]
+
+  filter {
+    name   = "name"
+    values = [each.value]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+}
+
+data "aws_iam_policy_document" "ad_fixngo" {
+  for_each = local.ad_fixngo.ec2_iam_policies
+
+  dynamic "statement" {
+    for_each = each.value.statements
+
+    content {
+      sid       = statement.value.sid
+      effect    = statement.value.effect
+      actions   = statement.value.actions
+      resources = statement.value.resources
+    }
+  }
+}
+
+resource "aws_iam_policy" "ad_fixngo" {
+  for_each = local.ad_fixngo.ec2_iam_policies
+
+  name        = each.key
+  path        = each.value.path
+  description = each.value.description
+  policy      = data.aws_iam_policy_document.ad_fixngo[each.key].json
+
+  tags = merge(local.tags, local.ad_fixngo.tags, {
+    Name = each.key
+  })
+}
+
+resource "aws_iam_role" "ad_fixngo" {
+  for_each = local.ad_fixngo.ec2_iam_roles
+
+  name                 = each.key
+  path                 = "/"
+  max_session_duration = "3600"
+  assume_role_policy = jsonencode(
+    {
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Effect" : "Allow",
+          "Principal" : {
+            "Service" : "ec2.amazonaws.com"
+          }
+          "Action" : "sts:AssumeRole",
+          "Condition" : {}
+        }
+      ]
+    }
+  )
+
+  managed_policy_arns = [
+    for key_or_arn in each.value.managed_policy_arns : try(aws_iam_policy.ad_fixngo[key_or_arn].arn, key_or_arn)
+  ]
+
+  tags = merge(local.tags, local.ad_fixngo.tags, {
+    Name = each.key
+  })
+}
+
+resource "aws_iam_instance_profile" "ad_fixngo" {
+  for_each = local.ad_fixngo.aws_instances
+
+  name = "ec2-profile-${each.key}"
+  role = each.value.iam_instance_profile_role
+  path = "/"
+}
+
+resource "aws_instance" "ad_fixngo" {
+  for_each = local.ad_fixngo.aws_instances
+
+  ami                    = data.aws_ami.ad_fixngo[each.value.ami_name].id
+  availability_zone      = each.value.availability_zone
+  ebs_optimized          = true
+  iam_instance_profile   = aws_iam_instance_profile.ad_fixngo[each.key].name
+  instance_type          = each.value.instance_type
+  private_ip             = each.value.private_ip
+  subnet_id              = each.value.subnet_id
+  user_data              = base64encode(file("./files/ad-fixngo-ec2-user-data.yaml"))
```

Comment:
Not really because the user-data script will join the server to the domain.  It's actually much less hassle overall to use base image here.  The user-data just calls a powershell script in modernisation-platform-configuration-management repo in similar way to ansible on linux instances.

Reply:
Yep, fair enough,